tox.ini

```diff
@@ -59,17 +59,56 @@ commands =
 deps =
     -r{toxinidir}/requirements/quality.txt
 commands =
-    black --check --diff --line-length=80 src/scriv tests docs setup.py
+    black --check --diff --line-length=80 src/scriv tests docs
     python -m cogapp -cP --check --verbosity=1 docs/*.rst
     mypy src/scriv tests
-    pylint src/scriv tests docs setup.py
-    pycodestyle src/scriv tests docs setup.py
-    pydocstyle src/scriv tests docs setup.py
-    isort --check-only --diff -p scriv tests src/scriv setup.py
-    python setup.py -q sdist
+    pylint src/scriv tests docs
+    pycodestyle src/scriv tests docs
+    pydocstyle src/scriv tests docs
+    isort --check-only --diff -p scriv tests src/scriv
+    python -m build --sdist
     twine check dist/*
 
 [testenv:upgrade]
 commands =
     python -m pip install -U pip
     make upgrade
+
+# Tools needed for running tests need to be configured in tox.ini instead of
+# pyproject.toml so that we can run tests without tomli installed on Pythons
+# before 3.11.
```

Comment:
Why? Isn't it behind an env marker anyway?

Reply:
You may also want to look into dependency groups. tox4 supports them natively. Although, `pip-tools` doesn't yet.

Reply:
Leaving the settings here works.  If you find a different way to run all the tox environments, we can consider changing it.

Reply:
I guess I don't fully understand the context of this code comment, then. I feel like it may be misleading because I don't think that the implication of the relation with tomli is accurate in this context. So my concern isn't with moving things into different config but rather rephrasing this, perhaps.

Reply:
I've added more to the comment.